engine/src/main/java/graphql/nadel/engine/execution/NadelExecutionStrategy.java

```diff
@@ -258,11 +258,11 @@ private static boolean fieldMatchesDefinition(GraphQLFieldDefinition graphQLFiel
     }
 
     private static boolean serviceOwnsNamespacedField(GraphQLObjectType namespacedObjectType, Service service) {
-        final boolean serviceExtendsNamespaceType = service.getDefinitionRegistry().getDefinitions(ObjectTypeExtensionDefinition.class)
+        return service.getDefinitionRegistry().getDefinitions(ObjectTypeDefinition.class)
                 .stream()
+                // the type can't be an extension in the owning service
+                .filter(objectTypeDef -> !(objectTypeDef instanceof ObjectTypeExtensionDefinition))
```

Comment:
Ah, I guess that's why you did the no object ext instead.

Reply:
Yep